### DISCUSSION TITLE
Font Awesome

### DISCUSSION THREAD

**Discussion by urshar** [liedekef/events-made-easy#31] [General]
I installed Font awesome Pro in WordPress, but I can't get the icons into the event description.
In the case of a post news, I have the option of inserting the icons using select.
What do I have to set so that I also have this option in the event description.

Andreas

**Comment by liedekef** [comment]
The plugin "Font Awesome" uses this code to check if their chooser should be enabled on a specific admin page:

```
        private function should_icon_chooser_be_enabled( $screen_id ) {
                if ( ! is_string( $screen_id ) ) {
                        return false;
                }

                return false !== array_search( $screen_id, $this->icon_chooser_screens, true );
        }       
```

The variable $this->icon_chooser_screens has as fixed value:
`        protected $icon_chooser_screens = array( 'post.php', 'post-new.php' );`

They use the outcome of that function to check if they should hook into tinymce to add the extra button.
In other words: only on post/new post pages their plugin adds the button to choose, with no way to add custom admin pages to it.
I tried quickly to change their function should_icon_chooser_be_enabled to just return "true" and the button started showing up just fine. In other words: they should provide a method/filter to allow other plugins to hook into the decision to show the button (or just be a little bit more intelligent in their decision: the filter to hook into tinymce can be added just fine everywhere in the backend).
So go back to the developers of that plugin and ask them for that, or use another plugin :-)

**Comment by robmadole** [comment]
Being able to use the button in someplace other than posting is a new use case for us. So you can't be too surprised we don't support it yet. @liedekef how would you envision a hook that would add additional screen_id's? Have you ran into this with other plugins?

**Reply by liedekef** [reply]
Any plugin can create custom admin pages (many do so, but less now since the custom post types exist). However mine still does :-)
Since you plug into tinymce I don't see the need to limit that to posts (why not all of admin screens?). But if you want to allow plugins to hook into this chooser, I would just do this:
```
        private function should_icon_chooser_be_enabled( $screen_id ) {
                if ( ! is_string( $screen_id ) ) {
                        return false;
                }
                $icon_chooser_screens = apply_filters('fa_icon_chooser_screens', $this->icon_chooser_screens);
                return false !== array_search( $screen_id, $icon_chooser_screens, true );
        }
```
So then people can create in their theme functions.php an add_filter on 'fa_icon_chooser_screens' and add additional screens to the array. Problem here is that this puts the dev-work on the user ... Of course plugins could do this too, but many plugins don't know (or want to know) that another plugin is installed (certainly not to add specific things).
I use wp_editor which renders the tinymce, so maybe you can find a way to add the button without people needing to do anything? Like I said, just making should_icon_chooser_be_enabled return true works fine for me (but maybe that has side-effects you don't want? but since the hook is in tinymce I don't see a downside ...).
